roseau/load_flow/utils/types.py

```diff
@@ -170,19 +169,31 @@ class InsulatorType(StrEnum):
     """Alias -- Medium-Density PolyEthylene (MDPE) insulation."""
 
     @classmethod
-    def _missing_(cls, value: object) -> "InsulatorType | None":
+    def _missing_(cls, value: object) -> "Insulator":
         if isinstance(value, str):
-            string = value.upper()
-            if string in {"", "NAN"}:
-                return cls.UNKNOWN
             try:
-                return cls[string]
+                return cls[value.upper()]
             except KeyError:
                 pass
-        msg = f"{value!r} cannot be converted into a InsulatorType."
+        msg = f"{value!r} cannot be converted into a Insulator."
         logger.error(msg)
         raise RoseauLoadFlowException(msg, RoseauLoadFlowExceptionCode.BAD_INSULATOR_TYPE)
 
     def code(self) -> str:
         """A code that can be used in insulator type names."""
         return self.name
+
+
+#
+# Deprecated enum classes
+#
+@deprecated("Use the class `Insulator` instead.")
+def InsulatorType(*args, **kwargs):  # noqa: N802
+    """A deprecated class: please use the class `Insulator` instead."""
+    return Insulator(*args, **kwargs)
+
+
+@deprecated("Use the class `Material` instead.")
+def ConductorType(*args, **kwargs):  # noqa: N802
+    """A deprecated class: please use the class `Material` instead."""
+    return Material(*args, **kwargs)
```

Comment:
These do not work because now `ConductorType.CU` will raise an exception. I think we can clean this file and add the following to the end of `roseau/load_flow/__init__.py`

```python
def __getattr__(name: str) -> Any:
    deprecated_classes = {"ConductorType": Material, "InsulatorType": Insulator}

    if name in deprecated_classes:
        import warnings

        from roseau.load_flow.utils._exceptions import find_stack_level

        new_class = deprecated_classes[name]
        warnings.warn(
            f"The {name} class is deprecated. Use {new_class.__name__} instead.",
            stacklevel=find_stack_level(),
        )
        return new_class
    else:
        # raise AttributeError with original error message
        raise AttributeError(f"module {__name__!r} has no attribute {name!r}")

```